airflow/jobs/scheduler_job.py

```diff
@@ -289,8 +290,21 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
             .filter(not_(DM.is_paused))
             .filter(TI.state == TaskInstanceState.SCHEDULED)
             .options(selectinload('dag_model'))
-            .order_by(-TI.priority_weight, DR.execution_date)
         )
+        if session.bind.dialect.name == "postgresql":
+            query = query.order_by(
+                TI.priority_weight.desc(),
+                TI.last_scheduling_decision.desc().nullsfirst(),
```

Comment:
Don't we want to look in asc order with nullsfirst?

Reply:
Using desc helps us go from the oldest date to recent. Using asc may make it difficult to reach a dag that first hit the max_active_runs

Reply:
Yeah, I figured we wanted oldest first, which would be asc, no?